lib/tests/2wp.js

```diff
@@ -902,6 +910,130 @@ const execute = (description, getRskHost) => {
+      await fundRskAccountThroughAPegin(rskTxHelper, btcTxHelper, senderRecipientInfo3.btcSenderAddressInfo);
+
+      const initial2wpBalances = await get2wpBalances(rskTxHelper, btcTxHelper);
+
+      const initialSender1AddressBalanceInSatoshis = await getBtcAddressBalanceInSatoshis(btcTxHelper, senderRecipientInfo1.btcSenderAddressInfo.address);
+      const initialSender2AddressBalanceInSatoshis = await getBtcAddressBalanceInSatoshis(btcTxHelper, senderRecipientInfo2.btcSenderAddressInfo.address);
+      const initialSender3AddressBalanceInSatoshis = await getBtcAddressBalanceInSatoshis(btcTxHelper, senderRecipientInfo3.btcSenderAddressInfo.address);
+
+      const pegout1ValueInSatoshis = MINIMUM_PEGOUT_AMOUNT_IN_SATOSHIS;
+      const pegout2ValueInSatoshis = MINIMUM_PEGOUT_AMOUNT_IN_SATOSHIS + 100_000;
+      const pegout3ValueInSatoshis = MINIMUM_PEGOUT_AMOUNT_IN_SATOSHIS + 150_000;
+      const totalPegoutValueInSatoshis = pegout1ValueInSatoshis + pegout2ValueInSatoshis + pegout3ValueInSatoshis;
+
+      const rskMemPoolTxHashes1 = await getRskMempoolTransactionHashes(rskTxHelper);
+      const initialRskMempoolTxHashesSize = rskMemPoolTxHashes1.length;
+
+      // Act
+
+      const shouldMine = false;
+      const pegoutTransaction1Promise = sendTxToBridge(rskTxHelper, new BN(satoshisToWeis(pegout1ValueInSatoshis)), senderRecipientInfo1.rskRecipientRskAddressInfo.address, shouldMine);
+      const pegoutTransaction2Promise = sendTxToBridge(rskTxHelper, new BN(satoshisToWeis(pegout2ValueInSatoshis)), senderRecipientInfo2.rskRecipientRskAddressInfo.address, shouldMine);
+      const pegoutTransaction3Promise = sendTxToBridge(rskTxHelper, new BN(satoshisToWeis(pegout3ValueInSatoshis)), senderRecipientInfo3.rskRecipientRskAddressInfo.address, shouldMine);
+      
+      // Waiting for our 3 txs to reach the mempool before trying to mine them. + 3 because we are sending 3 pegouts.
+      const atLeastExpectedCount = initialRskMempoolTxHashesSize + 3;
+      await waitForRskMempoolToGetAtLeastThisManyTxs(rskTxHelper, atLeastExpectedCount);
+
+      // Now our 3 pegout transaction requests will get mined together.
+      await rskTxHelper.mine();
+      
+      // After mining, we can get the transaction receipts.
+      const pegoutTransaction1 = await pegoutTransaction1Promise;
+      const pegoutTransaction2 = await pegoutTransaction2Promise;
+      const pegoutTransaction3 = await pegoutTransaction3Promise;
+
+      // Assert
+
+      let bridgeStateAfterPegoutCreation;
+      
+      // Callback to get the bridge state after the pegout is created
+      const pegoutCreatedCallback = async () => {
+        bridgeStateAfterPegoutCreation = await getBridgeState(rskTxHelper.getClient());
+      };
+
+      const callbacks = {
+        pegoutCreatedCallback
+      };
+
+      await triggerRelease(rskTxHelpers, btcTxHelper, callbacks);
+
+      // Checking all the pegout events are emitted and in order
```

Comment:
They all get mined in the same block, otherwise, the test would fail.
Since they can get mined in different order, I'm forced to use `pegoutsEvents.find` index of indices and `getAddressUtxo` to find the corresponding utxos.